packages/react-components/react-search/stories/SearchBox/SearchBoxControlled.stories.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { SearchBox } from '@fluentui/react-search';
+import { Field } from '@fluentui/react-components';
+import type { SearchBoxProps } from '@fluentui/react-search';
+
+export const Controlled = () => {
+  const [value, setValue] = React.useState('initial value');
+
+  const onChange: SearchBoxProps['onChange'] = (ev, data) => {
+    if (data.value.length <= 20) {
+      setValue(data.value);
```

Comment:
With this example since you're limiting the character output, it might be nice to demonstrate showing an error when the 20 character limit is reached and the user attempts to insert more text. That should be possible directly through the `Field` component's `validationState` and `validationMessage` props.